API-Demo/api_Index.js

```diff
@@ -0,0 +1,68 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const postsContainer = document.getElementById('posts-container');
+
+    const fetchPosts = async () => {
+        try {
+            const response = await fetch('https://jsonplaceholder.typicode.com/posts');
+            if (!response.ok) throw new Error('Network response was not ok');
+            const posts = await response.json();
+            displayPosts(posts);
+        } catch (error) {
+            console.error('Fetching posts failed:', error);
+            postsContainer.innerHTML = `<p class="error">Failed to load posts. Please try again later.</p>`;
+        }
+    };
+
+    const fetchComments = async (postId) => {
+        try {
+            const response = await fetch(`https://jsonplaceholder.typicode.com/posts/${postId}/comments`);
+            if (!response.ok) throw new Error('Network response was not ok');
+            return await response.json();
+        } catch (error) {
+            console.error(`Fetching comments for post ${postId} failed:`, error);
+            return [];
+        }
+    };
+
+    const displayPosts = async (posts) => {
+        const alternatePosts = posts.filter((_, index) => index % 2 === 0);
+        
+        const postsHtml = alternatePosts.map(({ id, title, body }) => `
+            <div class = "post-Number">Post - ${id} </div>
+            <div class="post" data-post-id="${id}">
+                <div class="post-header">
+                    <h2>${title}</h2>
+                    <button class="toggle-comments" data-post-id="${id}">Show Comments</button>
+                </div>
+                <p>${body}</p>
```

Comment:
no need to add $, javascript works if you write it inside curly brackets {}.
We use $ while creating a template string that involves variables

Reply:
Changes done.